### PR TITLE
Add support for HTML comments

### PR DESCRIPTION
**Pull request by tantalic**
This adds the ability to use HTML style comments, wrapped in` <!-- -->`. HTML comments are parsed into a new node type and included in the template output.

Basic design/approach followed #79. This is my first attempt to work on the templ project. I tried to follow the style/practices I see elsewhere. Happy to take any feedback/changes into this PR to move it along in the right direction for the project.

### CODE REVIEW

**Review by a-h — CHANGES_REQUESTED** [review]
Thanks for getting involved!

This looks like a great start, but some tests around it are important. They help to make sure that we don't break expected behaviour between releases.

There are two major types of test in templ. One tests that the parser behaves as expected - i.e. produces errors when it should, and handles edge cases. You can see an example of that in https://github.com/a-h/templ/blob/main/parser/v2/doctypeparser_test.go

The other is to test that the appropriate HTML output is generated - a more complete integration test.

In temp, do we expect the comments to be included, or should they be elided from the HTML output?

Either way, a test would show the expected behaviour and make sure we don't break it during refactoring or bug fixes. Here's an example of the generation style test - https://github.com/a-h/templ/tree/main/generator/test-templ-element

File: parser/v2/comment.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	"fmt"
+
+	"github.com/a-h/parse"
+)
+
+type commentParser struct {
+	start parse.Parser[string]
+	end   parse.Parser[string]
```
[the remark this answers is not in the document]
Comment (tantalic) [reply parser/v2/comment.go:11]: My thought here was that this struct could be extended to support `//` style comments as well. However, I’m not entirely sure yet if that is feasible or not yet, so curious on your thoughts there.

[the remark this answers is not in the document]
Comment (joerdav) [reply parser/v2/comment.go:11]: Personally I think that the implementation for Go style comments will be quite different, they won't arrive in the resulting HTML, but probably should end up in the generated Go code to support godoc style comments and build tag style comments.

File: parser/v2/templateparser.go [hunk 2]
```diff
@@ -88,6 +90,19 @@ func (p templateNodeParser[T]) Parse(pi *parse.Input) (op []Node, ok bool, err e
...
 
+		// Try for a comment.
+		// <!--
+		var commentNode Comment
+		commentNode, ok, err = comment.Parse(pi)
+		if err != nil {
+			return
+		}
+		if ok {
+			fmt.Println(commentNode.Contents)
```
[the remark this answers is not in the document]
Comment (tantalic) [reply parser/v2/templateparser.go:101]: I would have sworn I did a search for println calls specifically to avoid doing this. 🤦

**Comment by tantalic** [comment]
@a-h & @joerdav - Thank you for great feedback on this PR. I've addressed all the changes except for the tests which I will make some time to work on.

File: parser/v2/templateparser.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (a-h) [reply parser/v2/templateparser.go]: We've all done it!

**Comment by tantalic** [comment]
@a-h - I think this is ready for review again.

**Comment by a-h** [comment]
Thanks! Will take a look tomorrow. 🙌

**Review by a-h — APPROVED** [review]

**Comment by a-h** [comment]
Thanks a lot, that's great!
